File: javalibrary/src/main/java/com/bluthlee/javalibrary/ui/recorderWidget/FloatRecorderWidget.java
package com.bluthlee.javalibrary.ui.recorderWidget;

import android.app.Dialog;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;

import com.bluthlee.javalibrary.R;
import com.bluthlee.javalibrary.data.DataBinder;

/**
 * Created by LC on 2017/8/31.
 */

public class FloatRecorderWidget extends RecorderWidget {

    public View recorderView;

    public FloatRecorderWidget(Context context, DataBinder dataBinder) {
        super(dataBinder);
        dialog = new Dialog(context, R.style.FloatDialog);
        dialog.setCanceledOnTouchOutside(false);
        View view = LayoutInflater.from(context).inflate(R.layout.dialog_float, null, false);
        recorderView = view.findViewById(R.id.iv_recorder);
        view.findViewById(R.id.iv_close).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                setStatus(STATUS_CLOSE);
            }
        });
        dialog.setContentView(view);
        super.init();
    }

    @Override
    public void show() {
        super.show();
        dialog.show();
        setStatus(STATUS_IDLE);
    }

    @Override
    public void cancel() {
        super.cancel();
        dialog.cancel();
    }
}

File: kotlinlibrary/src/main/java/com/bluthlee/library/ui/recorderWidget/FloatRecorderWidget.kt
package com.bluthlee.library.ui.recorderWidget

import android.app.Dialog
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import com.bluthlee.library.data.DataBinder
import com.bluthlee.library.R

/**
 * Created by LC on 2017/8/25.
 */
internal class FloatRecorderWidget(context: Context, dataBinder: DataBinder) : RecorderWidget(dataBinder) {

    var recorderView: View

    init {
        dialog = Dialog(context, R.style.FloatDialog)
        dialog.setCanceledOnTouchOutside(false)
        val view = LayoutInflater.from(context).inflate(R.layout.dialog_float, null, false)
        recorderView = view.findViewById(R.id.iv_recorder)
        view.findViewById<ImageView>(R.id.iv_close).setOnClickListener {
            setStatus(STATUS_CLOSE)
        }
        dialog.setContentView(view)
        super.init()
    }

    override fun show() {
        dialog.show()
        setStatus(STATUS_IDLE)
    }

    override fun cancel() {
        dialog.cancel()
    }

}